.github/workflows/release.yml

```
@@ -67,53 +71,56 @@ jobs:
+          if ! [[ "${NPM_DIST_TAG}" =~ ^[a-z][a-z0-9._-]{0,127}$ ]]; then
+            echo "❌ Invalid npm dist-tag '${NPM_DIST_TAG}'. Must start with a lowercase letter and contain only [a-z0-9._-], max 128 chars." >&2
             exit 1
           fi
 
           # Disallow version-like prefixes (avoid semver/range confusion)
-          if [[ "$NPM_DIST_TAG" =~ ^v[0-9] || "$NPM_DIST_TAG" =~ ^[0-9] ]]; then
-            echo "❌ Invalid npm dist-tag '$NPM_DIST_TAG'. Must not start with 'v' + digit or a digit (e.g., 'v1', '1.2.3')." >&2
+          if [[ "${NPM_DIST_TAG}" =~ ^v[0-9] || "${NPM_DIST_TAG}" =~ ^[0-9] ]]; then
+            echo "❌ Invalid npm dist-tag '${NPM_DIST_TAG}'. Must not start with 'v' + digit or a digit (e.g., 'v1', '1.2.3')." >&2
             exit 1
           fi
 
-          echo "✅ npmjs_dist_tag '$NPM_DIST_TAG' is valid."
+          echo "✅ npmjs_dist_tag '${NPM_DIST_TAG}' is valid."
 
       - name: Get package version from package.json
         id: get_version
+        env:
+          PACKAGE_NAME_INPUT: ${{ github.event.inputs.package_name }}
+          NPMJS_DIST_TAG_INPUT: ${{ github.event.inputs.npmjs_dist_tag }}
         run: |
           set -euo pipefail
-          PACKAGE_NAME="${{ github.event.inputs.package_name }}"
+          PACKAGE_NAME="${PACKAGE_NAME_INPUT}"
           PKG_JSON="packages/${PACKAGE_NAME}/package.json"
-          if [[ ! -f "$PKG_JSON" ]]; then
-            echo "package.json not found at $PKG_JSON. Check 'package_name' input." >&2
+          if [[ ! -f "${PKG_JSON}" ]]; then
+            echo "package.json not found at ${PKG_JSON}. Check 'package_name' input." >&2
             exit 1
           fi
-          VERSION=$(jq -er .version "$PKG_JSON")
-          if [[ -z "$VERSION" || "$VERSION" == "null" ]]; then
-            echo "Version is empty or missing in $PKG_JSON" >&2
+          VERSION=$(jq -er .version "${PKG_JSON}")
+          if [[ -z "${VERSION}" || "${VERSION}" == "null" ]]; then
+            echo "Version is empty or missing in ${PKG_JSON}" >&2
             exit 1
           fi
-          NPM_DIST_TAG="${{ github.event.inputs.npmjs_dist_tag }}"
-          if [ -z "$NPM_DIST_TAG" ]; then
+          NPM_DIST_TAG="${NPMJS_DIST_TAG_INPUT}"
+          if [ -z "${NPM_DIST_TAG}" ]; then
             NPM_DIST_TAG="latest"
           fi
-          if [[ "$NPM_DIST_TAG" == "latest" ]] && ! [[ "$VERSION" =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
-            echo "With npmjs_dist_tag 'latest', version must be of the form x.y.z. Found '$VERSION'." >&2
+          if [[ "${NPM_DIST_TAG}" == "latest" ]] && ! [[ "${VERSION}" =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+            echo "With npmjs_dist_tag 'latest', version must be of the form x.y.z. Found '${VERSION}'." >&2
             exit 1
           fi
-          echo "PACKAGE_VERSION=$VERSION" >> "$GITHUB_ENV"
-          echo "version=$VERSION" >> "$GITHUB_OUTPUT"
+          echo "PACKAGE_VERSION=${VERSION}" >> "$GITHUB_ENV"
```

```suggestion
          echo "package_version=${VERSION}" >> "$GITHUB_OUTPUT"
```

In follow-up jobs you refer to `${{ needs.get_version.outputs.package_version }}` but that will always be empty, since you're writing the version as an env var and not as an output.